### PR TITLE
{SEMI-MODULAR} Silicon OOC notes and flavor text

### PR DESCRIPTION
**Pull request by NotRanged**
## About The Pull Request

Adds the OOC notes and flavor text functionality to silicons.
Tested locally and it all works.

## Why It's Good For The Game

dogborg erp obviously

## Changelog
:cl:
add: Added OOC notes and flavor text functionality for silicons.
/:cl:

### CODE REVIEW

**Comment by hugmeorelse** [comment]
Good feature. @Gandalf2k15 @Azarak @Useroth 

Please verify if this is up to standard.

File: code/modules/mob/living/silicon/examine.dm [hunk 1]
```diff
@@ -4,3 +4,16 @@
...
+		var/line
+		if(length(client.prefs.features["silicon_flavor_text"]))
+			var/message = client.prefs.features["silicon_flavor_text"]
+			if(length_char(message) <= 40)
+				line = "<span class='notice'>[message]</span>"
+			else
+				line = "<span class='notice'>[copytext_char(message, 1, 37)]... <a href='?src=[REF(src)];lookup_info=silicon_flavor_text'>More...</a></span>"
+		line += " <span class='notice'><a href='?src=[REF(src)];lookup_info=ooc_prefs'>\[OOC\]</a></span>"
+		. += line
+	// END SKYRAT EDIT
```
Comment (Azarak) [inline code/modules/mob/living/silicon/examine.dm:19]:
```suggestion
	//SKYRAT EDIT ADDITION BEGIN - Customization
	var/line
if(length(client.prefs.features["silicon_flavor_text"]))
		var/message = client.prefs.features["silicon_flavor_text"]
		if(length_char(message) <= 40)
			line = "<span class='notice'>[message]</span>"
		else
			line = "<span class='notice'>[copytext_char(message, 1, 37)]... <a href='?src=[REF(src)];lookup_info=silicon_flavor_text'>More...</a></span>"
	if(client)
		line += " <span class='notice'><a href='?src=[REF(src)];lookup_info=ooc_prefs'>\[OOC\]</a></span>"
	. += line
	//SKYRAT EDIT ADDITION END
```

Comment (Azarak) [reply code/modules/mob/living/silicon/examine.dm:19]: Man I hope I didnt mess up any indentitation. but basically the point is to show OOC prefs only when a client is present, but flavor text always.

File: code/modules/mob/living/silicon/silicon.dm [hunk 2]
```diff
@@ -210,6 +210,25 @@
 			return
 		to_chat(usr, href_list["printlawtext"])
 
+	// BEGIN SKYRAT EDIT: Silicon OOC Notes and Flavor Text
```
Comment (Azarak) [inline code/modules/mob/living/silicon/silicon.dm]:
```suggestion
	//SKYRAT EDIT ADDITION BEGIN - CUSTOMIZATION
```

File: code/modules/mob/living/silicon/silicon.dm [hunk 3]
```diff
@@ -210,6 +210,25 @@
...
+					popup.open()
+					return
+
+			if("silicon_flavor_text")
+				if(client && length(client.prefs.features["silicon_flavor_text"]))
+					var/datum/browser/popup = new(usr, "[name]'s flavor text", "[name]'s Flavor Text", 500, 200)
+					popup.set_content(text("<HTML><HEAD><TITLE>[]</TITLE></HEAD><BODY><TT>[]</TT></BODY></HTML>", "[name]'s flavor text", replacetext(client.prefs.features["silicon_flavor_text"], "\n", "<BR>")))
+					popup.open()
+					return
+	// END SKYRAT EDIT
```
Comment (Azarak) [inline code/modules/mob/living/silicon/silicon.dm]:
```suggestion
	//SKYRAT EDIT ADDITION END
```

File: code/modules/mob/living/silicon/examine.dm [hunk 4]
```diff
@@ -4,3 +4,16 @@
 		. += "<b>[src] has the following laws:</b>"
 		for(var/law in laws.get_law_list(include_zeroth = TRUE))
 			. += law
+
+	// BEGIN SKYRAT EDIT: OOC Notes for sillicons
```
Comment (Azarak) [inline code/modules/mob/living/silicon/examine.dm]:
```suggestion
	//SKYRAT EDIT ADDITION BEGIN - CUSTOMIZATION
```

File: code/modules/mob/living/silicon/examine.dm (hunk 1 above)
Comment (Azarak) [inline code/modules/mob/living/silicon/examine.dm]:
```suggestion
	//SKYRAT EDIT ADDITION END
```

**Review by Azarak — APPROVED** [review]

**Pull request merged**